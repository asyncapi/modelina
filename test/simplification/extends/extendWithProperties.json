{
  "type": "object",
  "properties": {
    "initial": {
      "type": "string"
    }
  },
  "allOf": [
    {
      "type": "object",
      "properties": {
        "street_address": {
          "type": "string"
        },
        "country": {
          "type": "string"
        }
      }
    },
    {
      "type": "object",
      "properties": {
        "street_address2": {
          "type": "string"
        },
        "country2": {
          "type": "string"
        }
      }
    }
  ]
}
