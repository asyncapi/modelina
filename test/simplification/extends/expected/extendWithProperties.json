[
  {
    "originalSchema":{
      "type":"object",
      "properties":{
        "initial":{
          "type":"string"
        }
      },
      "allOf":[
        {
          "type":"object",
          "properties":{
            "street_address":{
              "type":"string"
            },
            "country":{
              "type":"string"
            }
          }
        },
        {
          "type":"object",
          "properties":{
            "street_address2":{
              "type":"string"
            },
            "country2":{
              "type":"string"
            }
          }
        }
      ]
    },
    "type":"object",
    "$id":"anonymSchema1",
    "additionalProperties": {
      "originalSchema": true,
      "type": [
        "object",
        "string",
        "number",
        "array",
        "boolean",
        "null"
      ]
    },
    "properties":{
      "initial":{
        "originalSchema":{
          "type":"string"
        },
        "type":"string"
      }
    },
    "extend":[
      "anonymSchema2",
      "anonymSchema3"
    ]
  },
  {
    "originalSchema":{
      "type":"object",
      "properties":{
        "street_address":{
          "type":"string"
        },
        "country":{
          "type":"string"
        }
      }
    },
    "type":"object",
    "$id":"anonymSchema2",
    "additionalProperties": {
      "originalSchema": true,
      "type": [
        "object",
        "string",
        "number",
        "array",
        "boolean",
        "null"
      ]
    },
    "properties":{
      "street_address":{
        "originalSchema":{
          "type":"string"
        },
        "type":"string"
      },
      "country":{
        "originalSchema":{
          "type":"string"
        },
        "type":"string"
      }
    }
  },
  {
    "originalSchema":{
      "type":"object",
      "properties":{
        "street_address2":{
          "type":"string"
        },
        "country2":{
          "type":"string"
        }
      }
    },
    "type":"object",
    "$id":"anonymSchema3",
    "additionalProperties": {
      "originalSchema": true,
      "type": [
        "object",
        "string",
        "number",
        "array",
        "boolean",
        "null"
      ]
    },
    "properties":{
      "street_address2":{
        "originalSchema":{
          "type":"string"
        },
        "type":"string"
      },
      "country2":{
        "originalSchema":{
          "type":"string"
        },
        "type":"string"
      }
    }
  }
]